### DISCUSSION TITLE
Sonoff THR320D with Sonoff THS01

### DISCUSSION THREAD

**Discussion by titou4307** [arendst/Tasmota#22019] [Q&A]
Hello
Since a long time (20 months ....) I have two THR320D with Sonoff THS01...

This couple not function under Tasmota and I have put this problem with low cost product who are dead...
So I have bought a new couple (THR320D + THS01) and :
Under firmware sonoff : all is OK
Under tasmota after flash it is not OK

I have test under esphome and all is OK too.....

Does this firmware Tasmota last version accept this material ??

Thanks for your helps

**Comment by sfromis** [comment]
Well, what specific issues are behind the words "not OK"?

And what exactly did you do to configure it?

Details matter a lot.

**Comment by titou4307** [comment]
Hi THR320D under Tasmota and with the THS01 is not ok for me : 

The display don't display anything
On the web interface of Tasmota there are no value of the THS01 like this : 

null °C
null %
null °C

I have use the auto configuration (THR320D with THS01 and put the GPIO 25 with SI7021....

Sorry for my explanation, I am french and not a geek of this technology

Claude (from FRance)

**Reply by sfromis** [reply]
If you in the console enable `Weblog 3`, and watch what happens, are you getting messages similar to this?
`DHT: Pin25 timeout waiting for pulse 0`

In that case, I'd suggest changing `DhtDelay`, to try to adapt to your THS01.
The default is `DhtDelay 400,30` but with my THR316D/THS01, using `DhtDelay 450,50` works better. Other values can be found by experimentation.

**Comment by titou4307** [comment]
Hello
Your solution give a good result so many thanks

I have noted if I restart the tasmota device, it not keep the consign "DhtDelay 450,50 " at the restarting....

DhtDelay 450,50 return to DhtDelay 400,30 ....

Have you an idea to perform ?

Best regard
